apps/prairielearn/src/pages/instructorAssessmentAccess/instructorAssessmentAccess.ts

```diff
@@ -19,7 +27,101 @@ router.get(
       { assessment_id: res.locals.assessment.id },
       AssessmentAccessRulesSchema,
     );
-    res.send(InstructorAssessmentAccess({ resLocals: res.locals, accessRules }));
+
+    const assessmentPathExists = await fs.pathExists(
+      path.join(
+        res.locals.course.path,
+        'courseInstances',
+        res.locals.course_instance.short_name,
+        'assessments',
+        res.locals.assessment.tid,
+        'infoAssessment.json',
+      ),
+    );
+
+    let origHash = '';
+    if (assessmentPathExists) {
+      origHash = sha256(
+        b64EncodeUnicode(
+          await fs.readFile(
+            path.join(
+              res.locals.course.path,
+              'courseInstances',
+              res.locals.course_instance.short_name,
+              'assessments',
+              res.locals.assessment.tid,
+              'infoAssessment.json',
+            ),
+            'utf8',
+          ),
+        ),
+      ).toString();
+    }
+
+    res.send(InstructorAssessmentAccess({ resLocals: res.locals, accessRules, origHash }));
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (!res.locals.authz_data.has_course_instance_permission_edit) {
+      throw new error.HttpStatusError(403, 'Access denied (must be an instructor)');
+    }
+
+    if (res.locals.course.example_course) {
+      throw new error.HttpStatusError(403, 'Access denied. Cannot make changes to example course.');
+    }
+
+    if (req.body.__action === 'edit_access_rules') {
+      const assessmentsPath = path.join(
+        res.locals.course.path,
+        'courseInstances',
+        res.locals.course_instance.short_name,
+        'assessments',
+        res.locals.assessment.tid,
+        'infoAssessment.json',
+      );
+      if (!(await fs.pathExists(assessmentsPath))) {
+        throw new error.HttpStatusError(400, 'infoAssessment.json does not exist');
+      }
+
+      const paths = getPaths(req, res);
+
+      const assessmentInfo = JSON.parse(await fs.readFile(assessmentsPath, 'utf8'));
+
+      const origHash = req.body.__orig_hash;
+
+      const assessmentInfoEdit = assessmentInfo;
+      assessmentInfoEdit.allowAccess = JSON.parse(req.body.assessment_access_rules);
+
+      const editor = new FileModifyEditor({
+        locals: res.locals,
+        container: {
+          rootPath: paths.rootPath,
+          invalidRootPaths: paths.invalidRootPaths,
+        },
+        filePath: assessmentsPath,
+        editContents: b64EncodeUnicode(JSON.stringify(assessmentInfoEdit, null, 2)),
+        origHash,
+      });
+
+      if (!editor.shouldEdit()) {
+        res.redirect(req.originalUrl);
+        return;
+      }
+
```

Comment:
See https://github.com/PrairieLearn/PrairieLearn/pull/9909, you can remove this.

Reply:
Removed in [1dfef07](https://github.com/PrairieLearn/PrairieLearn/pull/9934/commits/1dfef07527ed0296ff3cdfa7c58d85fce2304b97).